src/vs/workbench/contrib/debug/browser/callStackView.ts

```diff
@@ -227,6 +227,7 @@ export class CallStackView extends ViewletPanel {
 		}));
 
 		this._register(this.debugService.onDidNewSession(s => {
+			this._register(s.onDidChangeLabel(() => this.onCallStackChangeScheduler.schedule()));
```

Comment:
There is no need to refresh the whole tree when the label of the session changes.
it is enough just to re-render the debug session element.
So I would go for `this.tree.rerender(s)` so the change in the view is minimal